tracer/src/Datadog.Trace/DiagnosticListeners/AspNetCoreDiagnosticObserver.cs

```diff
@@ -689,6 +691,8 @@ private void OnHostingUnhandledException(object arg)
                 AspNetCoreRequestHandler.HandleAspNetCoreException(tracer, CurrentSecurity, rootSpan, httpContext, unhandledStruct.Exception);
             }
 
+            CoreHttpContextStore.Instance.Remove();
```

Comment:
I don't know if you want this here? 🤔 [I believe `OnHostingHttpRequestInStop` is still called after `OnHostingUnhandledException`](https://github.com/dotnet/aspnetcore/blob/e25c5acceb5a50a7c525d8d011233530e8bb9ce6/src/Hosting/Hosting/src/Internal/HostingApplicationDiagnostics.cs#L220), so you'll end up not having a context earlier than you expect if you remove it here

Reply:
good point I removed it! [4d5bb45](https://github.com/DataDog/dd-trace-dotnet/pull/6151/commits/4d5bb450ce1ae264a4ee7ff0a8902d60f970f108)

Reply:
Double checked, and it's indeed called after `OnHostingUnhandledException`